gammapy/image/tests/test_mask.py

```diff
@@ -16,7 +16,32 @@ def test_random_creation():
     assert excluded[0].size != 0
 
 
-@pytest.mark.xfail
+@requires_dependency('scipy')
 def test_distance_image():
-    pass
-    # TODO
+    mask = ExclusionMask.empty(nxpix=3, nypix=2)
+    skymap = mask.distance_image
+    distance = skymap.data
+    assert distance.shape == (2, 3)
+    assert_allclose(distance[0, 0],  -1.)
+    assert_allclose(distance[-1, 0], -2.)
+    assert_allclose(distance[0, -1], -2.23606798)
+    assert_allclose(distance[-1, -1], -2.82842712)
+
+    mask = ExclusionMask.empty(nxpix=3, nypix=2, fill=1.)
+    skymap = mask.distance_image
+    distance = skymap.data
+    assert distance.shape == (2, 3)
+    assert_allclose(distance[0, 0], 1e10)
+    assert_allclose(distance[-1, -1], 1e10)
+
+    data = np.ones((2, 3))
+    data[0, :2] = 0.
+
+    mask = ExclusionMask(data=data)
+    skymap = mask.distance_image
+    distance = skymap.data
+    assert distance.shape == (2, 3)
+    assert_allclose(distance[0, 0], -1.)
+    assert_allclose(distance[-1, 0], 1.)
+    assert_allclose(distance[0, -1], 1.)
+    assert_allclose(distance[-1, -1], 1.41421356)
```

Comment:
Again, because the test matrix is so small, you could just put this, which is less lines and IMO a little more readable.
```python
expected = [[-1, -1, 1], [1, 1, 1.41421356]]
assert_allclose(distance, expected)
```